app/forms/collection_form.rb

```diff
@@ -53,7 +55,8 @@ def save
   end
 
   def valid?
-    run_callbacks(:validation) { super && collection.map(&:valid?).all? }
```

Comment:
The valid? method was previously flawed in that it was considering its own validity in the equation - however, in rails 5 - its own validity comes into question because the errors for each item in the collection are stored in the errors object even though they are empty, but the valid check simply checks for zero length.

The previous version also called the callbacks twice

This change simply checks for validity on all of the sub forms